File: bug13_conditions_wrong_order/Main.java
import java.util.Base64;
import java.nio.ByteBuffer;

public class Main {

    /* Bug: conditions in wrong order — x >= 0 always fires first for positive numbers */
    static long score(int x) {
        if (x >= 0)         return 0x0001L;
        else if (x >= 100)  return 0x0010L;
        else if (x >= 1000) return 0x0100L;
        else                return 0x1000L;
    }

    static void run() {
        int[] inputs = {5, 150, 2000, 50, 500, 3000, 10, 200, 1500, 75};

        boolean allLow = true;
        for (int x : inputs) {
            if (score(x) != 0x0001L) { allLow = false; break; }
        }
        if (allLow) {
            System.err.println("ValueError: all scores are 0x0001: conditions are in wrong order, first branch always fires");
            System.exit(1);
        }

        long acc = 0x123456789ABCDEF0L;
        for (int i = 0; i < inputs.length; i++) {
            long s = score(inputs[i]);
            acc = (acc ^ (s * (long)(i + 1))) * 0xBF58476D1CE4E5B9L;
            acc &= 0xFFFFFFFFFFFFFFFFL;
            acc = ((acc << 5) | (acc >>> 59)) & 0xFFFFFFFFFFFFFFFFL;
        }
        ByteBuffer buf = ByteBuffer.allocate(8);
        buf.putLong(acc);
        System.out.println(Base64.getEncoder().encodeToString(buf.array()));
    }

    public static void main(String[] args) {
        run();
    }
}

File: bug13_conditions_wrong_order/main.py
import base64, struct

def score(x):
    if x >= 0:
        return 0x0001
    elif x >= 100:
        return 0x0010
    elif x >= 1000:
        return 0x0100
    else:
        return 0x1000

def run():
    inputs = [5, 150, 2000, 50, 500, 3000, 10, 200, 1500, 75]
    scores = [score(x) for x in inputs]
    if all(s == 0x0001 for s in scores):
        raise ValueError("all scores are 0x0001: conditions are in wrong order, first branch always fires")
    acc = 0x123456789ABCDEF0
    for i, x in enumerate(inputs):
        s = score(x)
        acc = (acc ^ (s * (i + 1))) * 0xBF58476D1CE4E5B9
        acc &= 0xFFFFFFFFFFFFFFFF
        acc = ((acc << 5) | (acc >> 59)) & 0xFFFFFFFFFFFFFFFF
    raw = struct.pack(">Q", acc)
    print(base64.b64encode(raw).decode())

run()
